app/controllers/locks_controller.rb

```diff
@@ -59,32 +58,30 @@ def format_errors(object)
     object.errors.messages.values.flatten.join("\n")
   end
 
-  def for_stage_lock?
-    case action_name
-    when 'create'
-      (params[:lock] || {})[:resource_type] == "Stage"
-    when 'destroy'
-      lock.resource_type == "Stage"
-    when 'index', 'destroy_via_resource'
-      false
-    else
-      raise 'Unsupported action'
-    end
+  def require_stage
+    @resource_class = "Stage"
```

Comment:
better set the actual class here `Stage` and then compare to `@resource_class.name` ? (instead of constantizing)